src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs

```diff
@@ -231,20 +239,34 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
             if (!_threadActs.TryGetValue(threadID, out var threadAct))
                 return false;
 
-            int regSize = sizeof(x86_thread_state64_t);
+            int stateFlavor;
+            int stateCount;
+            int regSize;
+            (stateFlavor, stateCount, regSize) = Architecture switch
+            {
+                Architecture.Arm64 => (Native.ARM_THREAD_STATE64, Native.ARM_THREAD_STATE64_COUNT, sizeof(arm_thread_state64_t)),
+                _ => (Native.x86_THREAD_STATE64, Native.x86_THREAD_STATE64_COUNT, sizeof(x86_thread_state64_t))
```

Comment:
This second switch should be `Architecture.X86` and the default case should be `_ =>throw new PlatformNotSupportedException`.

Reply:
Don't you mean "Architecture.X64"? or both X86/X64?  Currently .NET Core only supports X64 and ARM64 (mono may support an X86 iOS, but I don't know for sure).

Reply:
Ah, right.  This is also why it should be explicit and not the default case.

Reply:
Yep, `default` is for `Architecture.Amd64`. Surely I can change it, but please also note that this style is used in other places, for example:
https://github.com/microsoft/clrmd/blob/d0409daab70c27dcf4c8b8d39cf91c11ae0773e6/src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs#L370
https://github.com/microsoft/clrmd/blob/d0409daab70c27dcf4c8b8d39cf91c11ae0773e6/src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs#L229